src/devices/SDLJoypad/CMakeLists.txt

```diff
@@ -24,9 +24,15 @@ if(ENABLE_SDLJoypad)
                                                       YARP_sig
                                                       YARP_dev)
 
-  target_include_directories(yarp_SDLJoypad SYSTEM PRIVATE ${SDL_INCLUDE_DIR})
-  target_link_libraries(yarp_SDLJoypad PRIVATE ${SDL_LIBRARY})
-#   list(APPEND YARP_${YARP_PLUGIN_MASTER}_PRIVATE_DEPS SDL) (not using targets)
+  if(YARP_HAS_SDL2)
+    target_include_directories(yarp_SDLJoypad SYSTEM PRIVATE ${SDL2_INCLUDE_DIRS})
+    target_link_libraries(yarp_SDLJoypad PRIVATE ${SDL2_LIBRARIES})
```

Comment:
On my Windows machine, `${SDL2_LIBRARIES}` seems not be available.
I had to replace it with `SDL2::SDL2 SDL2::SDL2main` as suggested by vcpkg:

```
$ ./vcpkg.exe install sdl2:x64-windows
Computing installation plan...
The following packages are already installed:
    sdl2[core]:x64-windows
Package sdl2:x64-windows is already installed

Total elapsed time: 13.86 us

The package sdl2:x64-windows provides CMake targets:

    find_package(SDL2 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE SDL2::SDL2 SDL2::SDL2main)
```

Reply:
Unfortunatly I checked SDL build system, and apparently the exported CMake  variable depend on wheter SDL has been build with autotools (as I imagine the Debian package is built) or with CMake (as done in vcpkg), so it is a big mess...